robottelo/ui/locators.py

```diff
@@ -1,14 +1,35 @@
 # -*- encoding: utf-8 -*-
-# vim: ts=4 sw=4 expandtab ai
+"""Implements different locators for UI"""
 
-"""
-Implements different locators for UI
-"""
+import collections
+import logging
 
 from selenium.webdriver.common.by import By
 
 
-menu_locators = {
+LOGGER = logging.getLogger(__name__)
+
+
+class LocatorDict(collections.Mapping):
+    """This class will log every time an item is selected"""
+    def __init__(self, *args, **kwargs):
+        self.store = dict(*args, **kwargs)
+
+    def __getitem__(self, key):
+        item = self.store[key]
+        LOGGER.debug(
+            'Accessing locator "%s" by %s: "%s"', key, item[0], item[1]
```

Comment:
Nope! When you call `debug`, a `logging.LogRecord` object is created. This object stores both the logging formatting string and the objects that will be typecast and injected into the string. Why is this done? See [here](https://docs.python.org/3/library/logging.html#logging.LogRecord):

> This functionality can be used to inject your own values into a LogRecord at creation time.

In Python 3, the `logging.Formatter` class accepts a `style` argument. This argument is `'%'` by default, but can be set to `'{'` or `'$'` instead. Unfortunately, this functionality is not available in Python 2.